### ISSUE TITLE
Use expired-callback to expire captcha

### ISSUE THREAD

**Issue by TheSharpieOne** [VividCortex/angular-recaptcha#57]
I just looked at the API docs for recaptcha, it looks like they added an `expired-callback` attribute which expires and the user needs to solve a new CAPTCHA.
Currently, because this API attribute wasn't there previously, the solution depends on a timeout. It should be changed to use this new API method.

**Comment by brutalhonesty** [comment]
:+1:

**Comment by thatgirlcancode** [comment]
+1

**Comment by jcsena** [comment]
+1

**Comment by KalinKanev** [comment]
+1

**Comment by jpmckearin** [comment]
This is implemented but not yet documented. Looking [here](https://github.com/VividCortex/angular-recaptcha/blob/master/src/directive.js), there is a onExpire attribute on the directive. I'll submit a PR with the documentation change.

**Comment by TheSharpieOne** [comment]
@jpmckearin Yes, there is an implementation, I originally did it (sorry for not documenting it).  The problem is, when I did it, reCaptcha did not have the `expired-callback` attribute they have now.  Thus the [current implementation uses a timeout](https://github.com/VividCortex/angular-recaptcha/blob/da4e73c35cefb9b5721d972218fac36b7f4fac4f/src/directive.js#L53-L60) which is hard-coded to 2 mins (as at the time 2 mins was (and currently is) how long it takes for the session to expire is).  But this may change and this directive should not rely on the hard-coded value.  Instead, it should use the new `expired-callback` that reCaptcha provides to trigger the [validity](https://github.com/VividCortex/angular-recaptcha/blob/da4e73c35cefb9b5721d972218fac36b7f4fac4f/src/directive.js#L54-L56) and [end-user callback](https://github.com/VividCortex/angular-recaptcha/blob/da4e73c35cefb9b5721d972218fac36b7f4fac4f/src/directive.js#L59)

**Comment by jcsena** [comment]
The  solutions is send function is using expired-callback(google)
service.js
``` javascript
//service.js
create: function (elm, key, fn, expired, conf) {
                conf.callback = fn;
                conf.sitekey = key;
                conf['expired-callback'] = expired;
                return getRecaptcha().then(function (recaptcha) {
                    return recaptcha.render(elm, conf);
                });
            },
```
directive.js
``` javascript
//directive.js
function expired(){
if(angular.isDefined(attrs.onExpire)){
  scope.onExpire();
}
}

 vcRecaptcha.create(elm[0], key, callback, expired {
                        theme: scope.theme || attrs.theme || null,
                        tabindex: scope.tabindex || attrs.tabindex || null
                    })

```

**Comment by TheSharpieOne** [comment]
@jcsena that wouldn't work as well as you would expect. When reCaptcha calls the expire callback it would be outside of angular's knowledge and would require an additional digest to register the changes made in the callback. It would also break backwards compatibility as the current expire callback passes the local variable of the widget id. 

tl;dr That is not the angular way to do that.

**Comment by jcsena** [comment]
Mmm it is posible but, i implement this code with some modifications and this callback i can made the expire event and no problem

**Comment by TheSharpieOne** [comment]
This is not resolved, the PR that closed this only documented the existing feature.  This ticket is to update the existing feature to use the method that is now provided by recaptcha.
